starlite/middleware/session.py

```diff
@@ -0,0 +1,250 @@
+            scope: The ASGI connection scope.
+            send: The ASGI send function.
+            cookie_keys: Session cookie keys that are sent in the current request. It is required to expire all session
+                cookies from the current request and are replaced with new cookies in the upcoming response.
+
+        Returns:
+            None.
+        """
+
+        async def wrapped_send(message: "Message") -> None:
+            """A wrapper around the send function, declared in local scope to
+            use closure values.
+
+            Args:
+                message: An ASGI message.
+
+            Returns:
+                None
+            """
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(scope=message)
+                data = self.dump_data(scope.get("session"))
+                cookie_params = self.config.dict(exclude_none=True, exclude={"secret", "key"})
+                for i, datum in enumerate(data, start=0):
+                    headers.append(
+                        "Set-Cookie",
+                        Cookie(value=datum.decode("utf-8"), key=f"{self.config.key}-{i}", **cookie_params)
+                        .to_header()
+                        .replace("Set-Cookie: ", ""),
+                    )
+                # Cookies with the same key overwrite the earlier cookie with that key. To expire earlier session
+                # cookies, first check how many session cookies will not be overwritten in this upcoming response. If
+                # leftover cookies are greater than or equal to 1, that means older session cookies have to be expired
+                # and their names are in cookie_keys.
+                cookies_left = len(cookie_keys) - len(data)
+                if cookies_left >= 1:
+                    cookie_params = self.config.dict(exclude_none=True, exclude={"secret", "max_age", "key"})
+                    for cookie_key in cookie_keys[len(data) :]:
+                        headers.append(
+                            "Set-Cookie",
+                            Cookie(value="null", key=cookie_key, expires=0, **cookie_params)
+                            .to_header()
+                            .replace("Set-Cookie: ", ""),
+                        )
+            await send(message)
+
+        return wrapped_send
+
+    async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> None:
+        """
+        Args:
+            scope: The ASGI connection scope.
+            receive: The ASGI receive function.
+            send: The ASGI send function.
+
+        Returns:
+            None
+        """
+        if scope["type"] in self.config.scopes:
+            scope.setdefault("session", {})
+            connection = HTTPConnection(scope)
+            cookie_keys = sorted(key for key in connection.cookies if self.config.key in key)
+            if cookie_keys:
+                with contextlib.suppress(KeyError):
```

Comment:
why do you supress the exceptions here and below? This is where I would expect to see try / except.

Reply:
The session should remain empty if the exception occurs. When the user code will see an empty session, it must start authentication again to set new session.